research/gaia/gremlin/gremlin_core/src/process/traversal/step/source.rs

```diff
@@ -170,7 +170,14 @@ pub fn graph_step_from(
                 step.set_tags(gremlin_step.get_tags());
                 let mut ids = vec![];
                 for id in opt.ids {
-                    let id = ID::from_str(&id).unwrap();
+                    let id = if let Ok(uid) = ID::from_str(&id) {
+                        uid
+                    } else if let Ok(id) = i64::from_str(&id) {
+                        // in case that source id is negative
+                        id as ID
```

Comment:
You'd better use `ID::try_from(id)` instead of `id as ID`.

Reply:
Done

Reply:
ID::try_from(id) will fail if id is negative, thus I simply use id as ID here. Similarly for the following.